src/scripts/components/LogMessage.js

```diff
@@ -12,41 +12,62 @@ class LogMessage {
     this.rootElement = rootElement;
     this.rootElement.classList.add('log-message');
 
-    this.message = '';
-    this.type = 'default';
+    this.messageList = [];
 
     this.show = this.show.bind(this);
     this.hide = this.hide.bind(this);
+    this.addMessage = this.addMessage.bind(this);
     this.render = this.render.bind(this);
     this.buildView = this.buildView.bind(this);
+    this.registerListener = this.registerListener.bind(this);
+
+    this.registerListener();
   }
 
   show(message, type = 'error', time = 2000) {
-    this.message = message;
-    this.type = type;
-
-    this.render();
+    this.addMessage(message, type, time);
 
     this.rootElement.classList.add('log-message_show');
-
-    setTimeout(this.hide, time);
+    this.render();
   }
 
   hide() {
     this.rootElement.classList.remove('log-message_show');
   }
 
+  addMessage(message, type = 'error', time = 2000) {
+    const id = (this.messageList[this.messageList.length - 1]?.id + 1) || 0;
+
+    this.messageList = this.messageList.slice(-4);
```

Comment:
Why exactly slice with -4? Is it for the ability to see only the messages that fit the screen?

Reply:
I select 5 most relevant posts.